### PR TITLE
Introduce `MigrateMonsterCollection` action

### PR DESCRIPTION
**Pull request by moreal**
This pull request introduces `MigrationMonsterCollection` action to migrate `MonsterCollectionState` to `StakeState` for keeping staking period.

### CODE REVIEW

**Comment by ipdae** [comment]
After executing the migration, we will prevent the execute actions related to MonsterCollection.

File: Lib9c/Action/MigrateMonsterCollection.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            var currency = states.GetGoldCurrency();
+            Address collectionAddress = MonsterCollectionState.DeriveAddress(context.Signer, agentState.MonsterCollectionRound);
+            if (!states.TryGetState(collectionAddress, out Dictionary stateDict))
+            {
+                throw new FailedLoadStateException($"Aborted as the monster collection state failed to load.");
+            }
+
+            var monsterCollectionState = new MonsterCollectionState(stateDict);
+            var migratedStakeStateAddress = StakeState.DeriveAddress(context.Signer);
+            var migratedStakeState = new StakeState(migratedStakeStateAddress, monsterCollectionState.ReceivedBlockIndex);
```
[the remark this answers is not in the document]
Comment (moreal) [reply Lib9c/Action/MigrateMonsterCollection.cs:32]: I didn't care about it. I'll make it do nothing if the `StakeState` already exists.

[the remark this answers is not in the document]
Comment (moreal) [reply Lib9c/Action/MigrateMonsterCollection.cs:61]: Hello @ipdae, I resolved this part in [6321a5c](https://github.com/planetarium/lib9c/pull/983/commits/6321a5c6435290a94858917b877fc452387d562f)

File: Lib9c/Action/MigrateMonsterCollection.cs [hunk 2]
```diff
@@ -36,6 +36,11 @@ public override void LoadPlainValue(IValue plainValue)
         public override IAccountStateDelta Execute(IActionContext context)
         {
             var states = context.PreviousStates;
+            if (states.TryGetStakeState(context.Signer, out StakeState _))
+            {
+                return states;
```
Comment (ipdae) [inline Lib9c/Action/MigrateMonsterCollection.cs:41]: i think, it will be better throw exception.

**Review by ipdae — APPROVED** [review]

File: Lib9c/Action/MigrateMonsterCollection.cs (hunk 2 above)
Comment (moreal) [reply Lib9c/Action/MigrateMonsterCollection.cs]: I made it throw `InvalidOperationException` in [3292248](https://github.com/planetarium/lib9c/pull/983/commits/3292248f099fd3a6a45702de701b5bf9d51c688e)

**Pull request merged**